examples/nlu-assistant/README.md

```diff
@@ -0,0 +1,507 @@
+  - [1. Create your knowledge base](#1-create-your-knowledge-base)
+  - [2. Creating your Neural Network model](#2-creating-your-neural-network-model)
+    - [2.1 Understanding the main points of a model](#21-understanding-the-main-points-of-a-model)
+    - [2.2 Creating your own custom model](#22-creating-your-own-custom-model)
+    - [2.3 Creating your own custom data preprocessors](#23-creating-your-own-custom-data-preprocessors)
+      - [2.3.1 Creating a custom tokenizer](#231-creating-a-custom-tokenizer)
+      - [2.3.2 Creating a custom normalizer](#232-creating-a-custom-normalizer)
+    - [2.4 Using your own custom model](#24-using-your-own-custom-model)
+      - [2.4.1 Initializing the NLU with custom preprocessors](#241-initializing-the-nlu-with-custom-preprocessors)
+      - [2.4.2 Preparing the data](#242-preparing-the-data)
+      - [2.4.3 Model configuration and tunning](#243-model-configuration-and-tunning)
+      - [2.4.4 Training and evaluating the model](#244-training-and-evaluating-the-model)
+    - [2.5 Run the model](#25-run-the-model)
+  - [3. Defining the conversational flow](#3-defining-the-conversational-flow)
+  - [4. Using the Botonic NLU plugin](#4-using-the-botonic-nlu-plugin)
+
+## How to use this example
+
+1. From your command line, download the example by running:
+   ```bash
+   $ botonic new myNluAssistant nlu-assistant
+   ```
+2. `cd` into `myNluAssistant` directory that has been created.
+3. Run `botonic serve` to test it in your local machine.
+
+Additionally, you can try to add new examples in **src/nlu/utterances/en** and train the model again by running:
+
+```bash
+$ botonic train
+```
+
+Now you can test how this new model performs by running `botonic serve` again and try with new inputs.
+
+## Walk-through
+
+In this walk-through we will cover the most important steps that has been done in order to develop this example.
+
+#### 1. Create your knowledge base
+
+First of all, we need to define some training examples to feed our neural network (**NN**). To do so, we are going to create **6** different files, one for each intent. Under **src/nlu/utterances/en/** create the following files:
+
+- `Greeting.txt`
+- `Joke.txt`
+- `GetDirections.txt`
+- `Weather.txt`
+- `Thanks.txt`
+- `Farewell.txt`
+
+Now, inside every one of them, add several examples as follows:
+
+**src/nlu/utterances/en/GetDirections.txt**
+
+```
+how can i reach a train station?
+get directions to the nearest ATM
+how do I get home taking E120
+find the fastest route to Baxter Building 42nd street Madison Avenue Manhattan
+how to get to the Main Street avoiding heavy traffic
+...
+```
+
+Each example should be defined in single line and should be separated of the next one by a newline.
+
+> **Note:** Take into account that in order to train deep learning models, the more example the better your neural network will perform. For the sake of this example we have added 50 examples per intent, but they could be insufficient.
```

Comment:
**Note:** Take into account that in order to train deep learning models, the more examples we have the better the neural network will perform. For the sake of this example, we have added 50 examples per intent (but they could be insufficient).